Source/Libraries/FaultData/DataWriters/Emails/ConfigurationLoader.cs

```diff
@@ -113,38 +105,60 @@ private IEnumerable<string[]> LoadXDASettings(AdoDataConnection connection)
 
             return settingList.Select(setting => ToArray(setting.Name, setting.Value));
         }
-
-        private IEnumerable<string[]> LoadDataSourceSettings(AdoDataConnection connection)
+        protected virtual IEnumerable<string[]> LoadDataSourceSettings(AdoDataConnection connection)
         {
-            if (DataSourceEmailTypeID is null)
-                return Enumerable.Empty<string[]>();
-
-            TableOperations<TriggeredEmailDataSourceSetting> settingTable = new TableOperations<TriggeredEmailDataSourceSetting>(connection);
-            List<TriggeredEmailDataSourceSetting> settingList = settingTable.QueryRecordsWhere("TriggeredEmailDataSourceEmailTypeID = {0}", DataSourceEmailTypeID).ToList();
-
-            foreach (IGrouping<string, TriggeredEmailDataSourceSetting> grouping in settingList.GroupBy(setting => setting.Name))
-            {
-                if (grouping.Count() > 1)
-                    Log.Warn($"Duplicate record for setting {grouping.Key} detected.");
-            }
-
-            return settingList.Select(setting => ToArray(setting.Name, setting.Value));
+            return Enumerable.Empty<string[]>();
         }
 
-        private string[] ToArray(string key, string value) =>
-            key.Split('.').Concat(new[] { value }).ToArray();
+        protected string[] ToArray(string key, string value) =>
+        key.Split('.').Concat(new[] { value }).ToArray();
 
         #endregion
 
         #region [ Static ]
 
         // Static Fields
-        private static readonly ILog Log = LogManager.GetLogger(typeof(ConfigurationLoader));
+        protected static readonly ILog Log = LogManager.GetLogger(typeof(ConfigurationLoader));
 
         // Static Properties
         private static ConnectionStringParser<SettingAttribute, CategoryAttribute> ConnectionStringParser { get; } =
             new ConnectionStringParser<SettingAttribute, CategoryAttribute>();
 
         #endregion
     }
+    public class ConfigurationLoader<T> : ConfigurationLoader where T : DataSourceSetting, new()
+    {
+        #region [Constructors]
+        public ConfigurationLoader(int dataSourceEmailTypeID, Func<AdoDataConnection> connectionFactory)
+            : base(connectionFactory)
+        {
+            DataSourceEmailTypeID = dataSourceEmailTypeID;
+        }
+        #endregion
+        #region [Properties]
+        private int DataSourceEmailTypeID { get; }
+        #endregion
+        #region [Methods]
+
+        protected override IEnumerable<string[]> LoadDataSourceSettings(AdoDataConnection connection)
+        {
+            TableOperations<T> settingTable = new TableOperations<T>(connection);
+            string query;
+            if (typeof(T) == typeof(ScheduledEmailDataSourceSetting))
+                query = "ScheduledEmailDataSourceID = {0}";
+            else
+                query = "TriggeredEmailDataSourceEmailTypeID = {0}";
```

Comment:
Hm, these fields don't seem to match. We should change `ScheduledEmailDataSourceSetting` so that it references `ScheduledEmailDataSourceEmailType` instead.

Also, the type comparison feels bad to me. Maybe we could go with this code to get the field name?

```c#
string dataSourceEmailTypeIDFieldName = settingTable
    .GetFieldNames()
    .FirstOrDefault(name => name.EndsWith("DataSourceEmailTypeID"));
```